internal/hcl/graph_vertex_data.go

```diff
@@ -73,15 +73,8 @@ func (v *VertexData) evaluate(e *Evaluator, b *Block) error {
 		return fmt.Errorf("data block %s has no label", v.ID())
 	}
 
-	var existingVals map[string]cty.Value
-	existingCtx := e.ctx.Get(b.TypeLabel())
-	if !existingCtx.IsNull() {
-		existingVals = existingCtx.AsValueMap()
-	} else {
-		existingVals = make(map[string]cty.Value)
-	}
-
-	val := e.evaluateResource(b, existingVals)
+	// We don't care about the existing values, this is only needed by the legacy evaluator
+	val := e.evaluateResource(b, map[string]cty.Value{})
```

Comment:
Is it possible if we can change this method `evaluateResource` to be something like `evaluateResourceOrDataBlock` (there's probably a better name than that). Then if we could add some comments to that method about the fact that the `values` argument isn't really needed for the graph evaluator. Just so we are clear in the future when we inevitably hit some bug e.t.c.